addons/auth_signup/controllers/main.py

```diff
@@ -42,6 +42,13 @@ def web_auth_signup(self, *args, **kw):
         if 'error' not in qcontext and request.httprequest.method == 'POST':
             try:
                 self.do_signup(qcontext)
+
+                # Set user to public if they were not signed in by do_signup
+                # (mfa enabled)
+                if request.session.uid is None:
+                    public_user = request.env.ref('base.public_user')
+                    request.update_env(user=public_user)
+
```

Comment:
The `request.env['ir.http']._auth_method_public()` method achieves this, it looks at `request.env.uid` instead of `request.session.uid` but in this context the two do the same. No need to change anything, just spreading the word that the method exists :)